airbase/airbase.py

```diff
@@ -137,23 +120,21 @@ def request(
         if country is None:
             country = self.all_countries
         else:
-            country = util.string_safe_list(country)
+            country = string_safe_list(country)
             self._validate_country(country)
 
         if pl is not None and shortpl is not None:
             raise ValueError("You cannot specify both 'pl' and 'shortpl'")
 
         # construct shortpl form pl if applicable
         if pl is not None:
-            pl_list = util.string_safe_list(pl)
-            shortpl = []
-            for p in pl_list:
-                try:
-                    shortpl.append(self.all_pollutants[p])
-                except KeyError:
-                    raise ValueError(
-                        "'{}' is not a valid pollutant name".format(p)
-                    )
+            pl_list = string_safe_list(pl)
+            try:
+                shortpl = [self.all_pollutants[p] for p in pl_list]
+            except KeyError as e:
+                raise ValueError(
+                    f"'{e.args[0]}' is not a valid pollutant name"
+                ) from None
```

Comment:
produces a "nicer" traceback

raising another exception gives you
``` python
try:
    {}["invalid_key"]
except Exception as e:
    raise ValueError(f"ups: {e}")
```
```
Traceback (most recent call last):
  File "lala.py", line 2, in <module>
    {}["invalid_key"]
KeyError: 'invalid_key'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "lala.py", line 4, in <module>
    raise ValueError(f"ups: {e}")
ValueError: ups: 'invalid_key'
```

raising `from e` is a bit better, but not what we want
``` python
try:
    {}["invalid_key"]
except Exception as e:
    raise ValueError(f"ups: {e}") from e
```
```
Traceback (most recent call last):
  File "lala.py", line 2, in <module>
    {}["invalid_key"]
KeyError: 'invalid_key'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "lala.py", line 4, in <module>
    raise ValueError(f"ups: {e}") from e
ValueError: ups: 'invalid_key'
```

raising `from None` is the shorter and hopefully the least confusing
``` python
try:
    {}["invalid_key"]
except Exception as e:
    raise ValueError(f"ups: {e}") from None
```
```
Traceback (most recent call last):
  File "lala.py", line 4, in <module>
    raise ValueError(f"ups: {e}") from None
ValueError: ups: 'invalid_key'
```

Reply:
Ah yes I remember this mechanism now. Personally I prefer the full traceback actually, they are often helpful in debugging, and otherwise since this isn't super common people generally know to look at the bottom for the most relevant part.

Reply:
> Personally I prefer the full traceback actually, they are often helpful in debugging, and otherwise since this isn't super common people generally know to look at the bottom for the most relevant part.

in that case I'll `raise ValueError(...) from e`

Reply:
Can you still implement this?

Reply:
sorry I forgot